tracex_project/trace_comparator/views.py

```diff
@@ -0,0 +1,184 @@
+from django.views.generic import FormView, TemplateView
+
+from extraction.models import PatientJourney
+from extraction.logic.orchestrator import Orchestrator, ExtractionConfiguration
+from trace_comparator.comparator import compare_traces
+from trace_comparator.forms import PatientJourneySelectForm
+from tracex.logic.utils import DataFrameUtilities as dfu
+
+import pandas as pd
+
+
+class TraceTestingOverviewView(FormView):
+    """View for selecting a patient journey for testing."""
+
+    form_class = PatientJourneySelectForm
+    template_name = "testing_overview.html"
+    success_url = reverse_lazy("journey_filter")
+
+    def form_valid(self, form):
+        """Pass selected journey to orchestrator."""
+        selected_journey = form.cleaned_data["selected_patient_journey"]
+        patient_journey_entry = PatientJourney.manager.get(name=selected_journey)
+        configuration = ExtractionConfiguration(
+            patient_journey=patient_journey_entry.patient_journey,
+        )
+        orchestrator = Orchestrator(configuration=configuration)
+        orchestrator.set_db_id_objects("patient_journey", patient_journey_entry.id)
+        self.request.session["patient_journey_name"] = selected_journey
+        self.request.session["is_comparing"] = True
+
+        return super().form_valid(form)
+
+
+class TraceTestingComparisonView(TemplateView):
+    """View for comparing the pipeline output against the ground truth."""
+
+    template_name = "testing_comparison.html"
+
+    def get_context_data(self, **kwargs):
+        """Preparing displaying pipeline extraction results"""
+        context = super().get_context_data(**kwargs)
+        patient_journey_name = self.request.session.get("patient_journey_name")
+        patient_journey = PatientJourney.manager.get(
+            name=patient_journey_name
+        ).patient_journey
+        pipeline_df = dfu.get_events_df(
+            patient_journey_name=patient_journey_name,
+            trace_position="last",
+        )
+
+        context.update(
+            {
+                "patient_journey_name": patient_journey_name,
+                "patient_journey": patient_journey,
+                "pipeline_output": pipeline_df.to_html(index=False),
+            }
+        )
+
+        return context
+
+    def get(self, request, *args, **kwargs):
+        """Return a JSON response with the current progress of the pipeline."""
+        is_ajax = request.headers.get("X-Requested-With") == "XMLHttpRequest"
+        if is_ajax:
+            progress_information = {
+                "progress": self.request.session.get("progress"),
+                "status": self.request.session.get("status"),
+            }
+            return JsonResponse(progress_information)
+
+        self.request.session["progress"] = 0
+        self.request.session["status"] = None
+        self.request.session.save()
+
+        return super().get(request, *args, **kwargs)
+
+    def post(self, request):
```

Comment:
So basically I need to execute logic when clicking on a button and being redirect afterwards. In Django there are 2 ways afaik. First would be using a form and second would be override the post function. Another solution would be JS but its weird to outsource part of the logic in JS when basically everything is in views. Workaround as you used before is to outsource the logic on the results page (what imo isn't that pretty, because of separation of concerns but maybe its Django practice idk). Another workaround would be to create an extra view with the logic and after executing it redirect to the results page.

Maybe you have a cleaner solution?

Reply:
I would also be very interested in a different approach, as I also only came up with the three mentioned by Kien and decided that overriding the post function would be best out of the three.

Reply:
I guess using a form is usually the preferred way to do this. But it can also be quite a hustle, especially if there already is another form used in the view. In this case we have a `TemplateView` changing that to `FormView` and using a form's submit button should work as an alternative for the post-override in this case.

Reply:
After talking to @FR-SON we will note this issuse in #93 down and exploring for other solutions later.